### ISSUE TITLE
Hide replies

### ISSUE THREAD

**Issue by marrus-sh** [tootsuite/mastodon#11984]
## Pitch:

Twitter is currently testing a feature to [hide replies](https://help.twitter.com/en/using-twitter/mentions-and-replies#hidden-reply). This does not (cannot, on Mastodon) actually *delete* posts, but it does keep them from showing up in the replies to a toot. It would be great if Mastodon had this as well (ideally, federated).

My guess is that you could signal this in a federated context with an ActivityStreams activity removing the post from the replies collection, but I haven't actually looked into it.

**Specific expected behaviours:** Hiding a reply should…

+ Prevent it from appearing in the replies to a toot for (at least) all followers of the original poster

+ Remove it from the notifications column for the poster

+ Automatically also hide any further replies to the hidden toot

+ Mute notifications for the hidden toot (so that the original poster doesn't get bothered about replies to a toot they have hidden)

**Bonus:** Replies from people you have blocked should be hidden automatically (upon being received).

## Motivation:

Uncurated comments feeds are bad, and the replies to a post are essentially a comments feed. If somebody says something unsavoury in reply to one of my posts, I don't want everyone who clicks to view the replies to have to see it.

**Comment by trwnh** [comment]
At the very least we should be able to hide certain replies from appearing on our permalinks.

**Comment by VyrCossont** [comment]
Very interested in this. Right now the only option users have is to ask a mod to delete replies from trolls, harassers, reply guys, etc. A self-service option would make it a whole lot easier.

**Comment by robsonsobral** [comment]
As replies can be a way to harassment, it should be possible to disable them completely or per toot bases.

**Comment by marksda1** [comment]
As a fairly new user I was surprised this is not an option. I just want to add another voice to this suggestion. I've had a couple of comments that are not worth a moderator's time but I wish there was an option to disassociate my post from the comments so other viewers don't see them. I guess my interim answer may be to delete and re-post at some point.

**Comment by bbbradsmith** [comment]
Something I've noticed is that if I block someone for making unwanted replies, even though I can no longer see their replies, everyone else will still see them if they visit that thread.

I assume the blocked user won't be able to make further comments, but it seems odd that you can't remove the replies that you blocked them for in the first place?

Perhaps we don't want the original poster to "own" the entire thread, and all replies should be accessible to those who want to see them, but  at least could the blocked user's replies be segregated or CW-hidden, or marked with "this reply is blocked by its recipient"? Something to allow other visitors to the thread to understand the context, that the recipient can't see the reply, and isn't making any tacit approval of it.

**Comment by trwnh** [comment]
>  at least could the blocked user's replies be segregated or CW-hidden, or marked with "this reply is blocked by its recipient"? 

#10916

**Comment by hasselmm** [comment]
One of the primary goals of Mastodon was it to stop online abuse and harassing. Moderating abusive comments is an essential tool for reaching this goal. Therefore I'd really like to see a solution for this problem. Even if it's hard to implement. Even if the implementation is imperfect and leaves non-trivial loopholes.

**Comment by yunkillable** [comment]
Super interested in this! Sometimes someone replies to a post and I'm on good terms with them but what they said is something I'm personally not about and don't want to see, but isn't anything rule breaking or worth breaking a friendship over (such as trying to start a debate about a TV show or making an innocent but ill-informed comment) so it would be nice to have the option to just hide that and not have it visible to build any resentment over or feel you're going to be tied to.
